### PR TITLE
[skip-ci] [ntuple] Add naming rules to the specification

### PR DESCRIPTION
**Pull request by vepadulano**

### CODE REVIEW

File: tree/ntuple/v7/doc/BinaryFormatSpecification.md [hunk 1]
```diff
@@ -1059,6 +1059,17 @@ The limits refer to a single RNTuple and do not consider combinations/joins such
 | Maximum string length (meta-data)              | 4GB                          | String encoding                                        |
 | Maximum RBlob size                             | 128 PiB                      | 1GiB / 8B * 1GiB (with maxKeySize=1GiB, offsetSize=8B) |
 
+## Naming specification
+
+The allowed character set for the strings representing the name of an RNTuple, as well as the name of a field, is
+restricted to printable Unicode characters encoded as UTF-8, with the following exceptions:
+
+* Full stop (U+002E, `.`)
+* Space (U+0020, ` `)
```
Comment (jblomer) [inline tree/ntuple/v7/doc/BinaryFormatSpecification.md:1068]: I'd say all white spaces (space, tab, small space etc.)

Comment (vepadulano) [reply tree/ntuple/v7/doc/BinaryFormatSpecification.md:1068]: Ok, let me check the full list and come back to this, I'd like to be as specific as possible.

File: tree/ntuple/v7/doc/BinaryFormatSpecification.md [hunk 2]
```diff
@@ -1059,6 +1059,17 @@ The limits refer to a single RNTuple and do not consider combinations/joins such
 | Maximum string length (meta-data)              | 4GB                          | String encoding                                        |
 | Maximum RBlob size                             | 128 PiB                      | 1GiB / 8B * 1GiB (with maxKeySize=1GiB, offsetSize=8B) |
 
+## Naming specification
+
+The allowed character set for the strings representing the name of an RNTuple, as well as the name of a field, is
+restricted to Unicode characters encoded as UTF-8, with the following exceptions:
+
+* All control codes. These notably include newline (U+000A) and horizontal tab (U+0009).
+* Full stop (U+002E '&#46;')
```
Comment (pcanal) [inline tree/ntuple/v7/doc/BinaryFormatSpecification.md:1068]: This is the UK formulation (`Period` is the US formulation).

**Review by pcanal — APPROVED** [review]

**Comment by jblomer** [comment]
@vepadulano Please feel free to rebase & merge. Please also submit a backport PR for v6.34.

**Comment by enirolf** [comment]
Just to point out a related PR: #16836. Following the specification proposed here empty field names are allowed, which technically is true to begin with since we use it for the zero field and cardinality fields. The PR linked here reflects this.

File: tree/ntuple/v7/doc/BinaryFormatSpecification.md (hunk 2 above)
Comment (vepadulano) [reply tree/ntuple/v7/doc/BinaryFormatSpecification.md:1068]: Thanks, I just took it from https://en.wikipedia.org/wiki/List_of_Unicode_characters, in any case the Unicode code disambiguates :+1:

**Comment by vepadulano** [comment]
I have also added in the formalisation the fact that empty names are not allowed on disk

**Pull request merged**